nautobot/docs/models/extras/dynamicgroup.md

```diff
@@ -1,7 +1,166 @@
+
+The Dynamic Group edit view has a **Filter Fields** tab that allows one to specify filter criteria. The filter fields available for a given Content Type are backed and validated by underlying filterset classes (for example `nautobot.dcim.filters.DeviceFilterSet`) and are represented in the web interface as a dynamically-generated filter form that corresponds to each eligible filter field.
+
+Any invalid field names that are not eligible for filtering objects will be discarded upon validation. Any invalid field values will result in a validation error.
+
+## Advanced Filtering
+
+<!-- markdownlint-disable MD036 -->
+_Added in version 1.4.0_
+<!-- markdownlint-enable MD036 -->
+
+Advanced filtering is performed using nested Dynamic Group memberships.
+
+An object is considered a member of an advanced Dynamic Group if it matches the aggregated filter criteria across all descendant groups.
+
+The Dynamic Group edit view has a **Child Groups** tab that allows one to make other Dynamic Groups of the same content type children of the parent group.
+
+!!! important
+    Filter fields and child groups are mutually exclusive. A group may have either a filter defined, or child groups, but not both.
+
+### Filter Generation
+
+Descendant filters are always processed from the top down (or from left to right) starting with the parent group and ending with the last nested child group.
+
+The nesting of Dynamic Group is performed using two advanced patterns: Set and graphs. Rules for each child group are processed using a set `operator`, and groups are sorted hierarchically as a directed acyclic graph (DAG), where the `weight` is used for sorting child groups topologically.
+
+In both cases, the ordering of the tree of descendants from a parent group to its nested children is significant and critically important to how each subsequent filter or group of filters are processed to result in a final set of member objects.
+
+Consider an example where there is a graph from the parent group to three direct child groups, the third of which has its own nested child group:
+
+```no-highlight
+parent
+- first-child
+- second-child
+- third-child
+  - nested-child
+```
+
+The filter generation would walk the graph topologically, starting from the base filter of `parent`, the filter of `first-child` would be applied to that of `parent`, then `second-child`, in order, all the way down. In the case of `third-child`, all of its children (only `nested-child` in this case) would be processed in order in the same way and the resultant filter from all of the child groups for `third-child` would be used to process the filter that resulted from the filter of `second-child`.
+
+If this is confusing, don't worry. We'll cover it more in hands-on examples after this section.
```

Comment:
Before jumping into this, there should be an example setup with each group's rules defined so it's clear on what exactly we're working with, and then introduce the more complex concepts iteratively.

Reply:
Ugh this has turned out to be super confusing to jump back and forth from technical reference to user guide. I decided to split them up.